include/rocksdb/db.h

```diff
@@ -214,8 +232,18 @@ class DB {
   // Open the database for read only.
   //
   static Status OpenForReadOnly(const Options& options, const std::string& name,
-                                DB** dbptr,
+                                std::unique_ptr<DB>* dbptr,
```

Comment:
Since we're changing the signature, maybe move `dbptr` to the end per coding guidelines.

Reply:
`error_if_wal_file_exists` is an optional parameter, so it has to be last if we keep it.